include/pybind11/pytypes.h

```diff
@@ -1784,14 +1784,11 @@ class kwargs : public dict {
     PYBIND11_OBJECT_DEFAULT(kwargs, dict, PyDict_Check)
 };
 
-class set : public object {
+class set_base : public object {
+protected:
```

Comment:
I don't like having add as a public function on frozen-sets. We can still call the C-API directly on the pointer for casters, but other code shouldn't be able to add objects to frozen sets.

Reply:
moved non-const methods to protected

Reply:
They would still need to be public for normal set?

Reply:
@ecatmur Why not just move them to set subclass and use the raw C-API to interact with the frozen set.

Reply:
Sorry, I get it now. Fixed.